nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core/Discover/SdkProjectDiscovery.cs

```diff
@@ -1,129 +1,347 @@
-                    dependencies = dependencies
-                        .Select(d => d with { TargetFrameworks = tfms })
-                        .ToImmutableArray();
-                    var transitiveDependencies = await GetTransitiveDependencies(repoRootPath, projectPath, tfms, dependencies, logger);
-                    ImmutableArray<Dependency> allDependencies = dependencies.Concat(transitiveDependencies).Concat(sdkDependencies)
-                        .OrderBy(d => d.Name)
-                        .ToImmutableArray();
-
-                    results.Add(new()
-                    {
-                        FilePath = Path.GetRelativePath(workspacePath, buildFile.Path),
-                        Properties = properties,
-                        TargetFrameworks = tfms,
-                        ReferencedProjectPaths = referencedProjectPaths,
-                        Dependencies = allDependencies,
-                    });
+                    File.Delete(binLogPath);
                 }
-                else
+                catch
                 {
-                    results.Add(new()
-                    {
-                        FilePath = Path.GetRelativePath(workspacePath, buildFile.Path),
-                        Dependencies = directDependencies.Concat(sdkDependencies)
-                            .OrderBy(d => d.Name)
-                            .ToImmutableArray(),
-                    });
                 }
             }
         }
 
-        return results.ToImmutable();
+        // and done
+        var projectDiscoveryResults = packagesPerProject.Keys.Select(projectPath =>
+        {
+            // gather some project-level information
+            var packagesByTfm = packagesPerProject[projectPath];
+            var projectFullDirectory = Path.GetDirectoryName(projectPath)!;
+            var doc = XDocument.Load(projectPath);
+            var localPropertyDefinitionElements = doc.Root!.XPathSelectElements("/Project/PropertyGroup/*");
+            var projectPropertyNames = localPropertyDefinitionElements.Select(e => e.Name.LocalName).ToHashSet(StringComparer.OrdinalIgnoreCase);
+            var projectRelativePath = Path.GetRelativePath(workspacePath, projectPath);
+            var topLevelPackageNames = topLevelPackagesPerProject.GetOrAdd(projectPath, () => new(StringComparer.OrdinalIgnoreCase));
+
+            // create dependencies
+            var tfms = packagesByTfm.Keys.OrderBy(tfm => tfm).ToImmutableArray();
+            var dependencies = tfms.SelectMany(tfm =>
+            {
+                return packagesByTfm[tfm].Keys.Select(packageName =>
+                {
+                    var packageVersion = packagesByTfm[tfm][packageName]!;
+                    var isTopLevel = topLevelPackageNames.Contains(packageName);
+                    var dependencyType = isTopLevel ? DependencyType.PackageReference : DependencyType.Unknown;
```

Comment:
Will this deal with `GlobalPackageReference` items, or by the time MSBuild is processing it it'll have been turned into a normal `PackageReference`?

Reply:
`GlobalPackageReference` is turned into a `PackageReference` so we get this for free, but I've added a test with explicitly this scenario; good catch!